src/lib.rs

```diff
@@ -208,9 +208,108 @@
 // Ensure we're `no_std` when compiling for Wasm.
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use srml_support::traits::Currency;
-use srml_support::{decl_event, decl_module, decl_storage, dispatch, StorageValue};
-use system::ensure_signed;
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate serde_derive;
+
+use rstd::prelude::*;
+
+use parity_codec_derive::{Decode, Encode};
+use srml_support::{ decl_event, decl_module, decl_storage, ensure, dispatch, StorageValue, StorageMap};
+
+/// Length constraint for input validation
+
+enum LengthValidationResult {
+    TooShort,
+    TooLong,
+    Success
+}
+
+struct InputValidationLengthConstraint {
+
+    /// Minimum length
+    min : usize,
+
+    /// Maximum length
+    /// While having max would have been more direct, this
+    /// way makes max < min unrepresentable semantically, 
+    /// which is safer.
+    max_as_min_diff: usize,
+}
+
+impl InputValidationLengthConstraint {
+    
+    /// Helper for computing max
+    fn max(&self) -> usize {
+        self.min + self.max_as_min_diff
+    }
+
+    /// Just to give method interface to read min, like max
+    fn min(&self) -> usize {
+        self.min
+    }
+
+    fn validate(&self, length: usize) -> LengthValidationResult {
+
+        if length < self.min {
+            LengthValidationResult::TooShort
+        }
+        else if length > self.max() {
+            LengthValidationResult::TooLong
+        }
+        else {
+            LengthValidationResult::Success
+        }
+    } 
+
+    // TODO: add more vliadtion stuff here in the future?
+}
+
+/// Constants
+const CATEGORY_TITLE: InputValidationLengthConstraint = InputValidationLengthConstraint{
+    min: 3,
+    max_as_min_diff: 30
+};
+
+const CATEGORY_DESCRIPTION: InputValidationLengthConstraint = InputValidationLengthConstraint{
+    min: 10,
+    max_as_min_diff: 100
+};
+
+/// The greatest valid depth of a category.
+/// The depth of a root category is 0.
+const MAX_VALID_CATEGORY_DEPTH: u32 = 3;
+
+/// Error messages for dispatchables
+/// Later perhaps make error message functions, to add parametization
+
+const DISPATCHABLE_ERROR_FORUM_SUDO_NOT_SET: &str = "Forum sudo not set.";
```

Comment:
`DISPATCHABLE_` prefix looks redundant. `ERROR_` prefix is more than eough.
```suggestion
const ERROR_FORUM_SUDO_NOT_SET: &str = "Forum sudo not set.";
```

Reply:
I was thinking if we have other error messages that are not in dispatchable, then it would be nice to differentiate, but I think its unlikely, removing.